python_modules/dagster/dagster/_core/storage/fs_io_manager.py

```diff
@@ -199,6 +199,24 @@ def handle_output(self, context, obj):
 
     def load_input(self, context):
         """Unpickle the file and Load it to a data object."""
+        if context.has_asset_partitions:
+            return [
+                self._load_pickle(self._partition_path(context.asset_key, partition_key))
+                for partition_key in context.asset_partition_keys
+            ]
+        else:
+            return self._load_input(context)
+
+    def _partition_path(self, asset_key, partition_key):
+        return os.path.join(self.base_dir, *asset_key.path, partition_key)
+
+    def _load_pickle(self, filepath):
```

Comment:
If we're going to put this in its own function, I think it's worth using in the other site inside the class where we `pickle.load`.

Reply:
I agree, to DRY it up I've also updated the existing load logic to call the `_load_pickle` private method